### ISSUE TITLE
pfmerge invokes PFADD instead of PFMERGE

### ISSUE THREAD

**Issue by christophstrobl** [mp911de/lettuce#158]
`pfmerge` should operate normally as soon as keys map to the same slot. But only the first key is considered.

Using lettuce 3.3.1 the following fails.
```java
RedisAdvancedClusterConnection<String, String> connection = client.connectCluster();

assumeThat(SlotHash.getSlot("key2660") == SlotHash.getSlot("key7112"), is(true));
assumeThat(SlotHash.getSlot("key7112") == SlotHash.getSlot("key8885"), is(true));

connection.pfadd("key2660", "rand", "mat");
connection.pfadd("key7112", "mat", "perrin");

connection.pfmerge("key8885", "key2660", "key7112");

assertThat(connection.pfcount("key8885"), is(3L));
```

**Comment by mp911de** [comment]
Will take a look into it.

**Comment by mp911de** [comment]
Calling `PFMERGE` invokes currently `PFADD` in all versions (3.3, 3.4, 4.0 and 4.1). The signature of the `pfmerge` is method is `Long pfmerge(...)` but should be `String pfmerge(...)`.

**Comment by mp911de** [comment]
Fixed in 3.3 , 3.4 (master), 4.0 and 4.1 branches. 3.3.2-SNAPSHOT and 3.4-SNAPSHOT jars are available in https://oss.sonatype.org/content/repositories/snapshots repo.

**Comment by mp911de** [comment]
Released 3.3.2.Final and 4.0.2.Final to Maven central.
